jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/BuildStepsRunner.java

```diff
@@ -226,53 +226,36 @@ public JibContainer build(
 
       return jibContainer;
 
-    } catch (ExecutionException executionException) {
-      Throwable exceptionDuringBuildSteps = executionException.getCause();
+    } catch (HttpHostConnectException ex) {
+      // Failed to connect to registry.
+      throw new BuildStepsExecutionException(helpfulSuggestions.forHttpHostConnect(), ex);
 
-      if (exceptionDuringBuildSteps instanceof HttpHostConnectException) {
-        // Failed to connect to registry.
-        throw new BuildStepsExecutionException(
-            helpfulSuggestions.forHttpHostConnect(), exceptionDuringBuildSteps);
+    } catch (RegistryUnauthorizedException ex) {
+      handleRegistryUnauthorizedException(ex, helpfulSuggestions);
 
-      } else if (exceptionDuringBuildSteps instanceof RegistryUnauthorizedException) {
-        handleRegistryUnauthorizedException(
-            (RegistryUnauthorizedException) exceptionDuringBuildSteps, helpfulSuggestions);
-
-      } else if (exceptionDuringBuildSteps instanceof RegistryCredentialsNotSentException) {
-        throw new BuildStepsExecutionException(
-            helpfulSuggestions.forCredentialsNotSent(), exceptionDuringBuildSteps);
+    } catch (RegistryCredentialsNotSentException ex) {
+      throw new BuildStepsExecutionException(helpfulSuggestions.forCredentialsNotSent(), ex);
 
-      } else if (exceptionDuringBuildSteps instanceof RegistryAuthenticationFailedException
-          && exceptionDuringBuildSteps.getCause() instanceof HttpResponseException) {
-        RegistryAuthenticationFailedException failureException =
-            (RegistryAuthenticationFailedException) exceptionDuringBuildSteps;
+    } catch (RegistryAuthenticationFailedException ex) {
+      if (ex.getCause() instanceof HttpResponseException) {
```

Comment:
What happens if the cause isn't a `HttpResponseException`?

Reply:
Ah right, I'll change it to match previous behavior.